app/src/test/java/org/pesmypetcare/mypetcare/controllers/exercise/TestTrUpdateExercise.java

```diff
@@ -0,0 +1,138 @@
+        trUpdateExercise.setExerciseName("Frisbee");
+        trUpdateExercise.setExerciseDescription("Playing at the beach");
+        trUpdateExercise.setOriginalStartDateTime(DateTime.Builder.buildFullString("2020-05-04T10:00:00"));
+        trUpdateExercise.setStartDateTime(DateTime.Builder.buildFullString("2020-05-04T10:00:00"));
+        trUpdateExercise.setEndDateTime(DateTime.Builder.buildFullString("2020-05-04T11:00:00"));
+        trUpdateExercise.execute();
+    }
+
+    @Test(expected = InvalidPeriodException.class)
+    public void shouldStartAndEndTheExerciseInTheSameDate() throws NotPetOwnerException, InvalidPeriodException,
+            NotExistingExerciseException, ExecutionException, InterruptedException {
+        trUpdateExercise.setUser(user);
+        trUpdateExercise.setPet(pet);
+        trUpdateExercise.setExerciseName("Frisbee");
+        trUpdateExercise.setExerciseDescription("Playing at the beach");
+        trUpdateExercise.setOriginalStartDateTime(DateTime.Builder.buildFullString("2020-05-04T11:00:00"));
+        trUpdateExercise.setStartDateTime(DateTime.Builder.buildFullString("2020-05-04T11:00:00"));
+        trUpdateExercise.setEndDateTime(DateTime.Builder.buildFullString("2020-05-05T10:00:00"));
+        trUpdateExercise.execute();
+    }
+
+    @Test(expected = NotExistingExerciseException.class)
+    public void shouldNotUpdateNonExistingExercise() throws NotPetOwnerException, NotExistingExerciseException,
+            InvalidPeriodException, ExecutionException, InterruptedException {
+        trUpdateExercise.setUser(user);
+        trUpdateExercise.setPet(pet);
+        trUpdateExercise.setExerciseName("Frisbee2");
+        trUpdateExercise.setExerciseDescription("Playing at the beach");
+        trUpdateExercise.setOriginalStartDateTime(DateTime.Builder.buildFullString("2020-05-04T11:00:00"));
+        trUpdateExercise.setStartDateTime(DateTime.Builder.buildFullString("2020-05-04T12:00:00"));
+        trUpdateExercise.setEndDateTime(DateTime.Builder.buildFullString("2020-05-04T13:00:00"));
+        trUpdateExercise.execute();
+    }
+
+    @Test
+    public void shouldUpdateTheExercise() throws NotPetOwnerException, NotExistingExerciseException,
+            InvalidPeriodException, ExecutionException, InterruptedException {
+        trUpdateExercise.setUser(user);
+        trUpdateExercise.setPet(pet);
+        trUpdateExercise.setExerciseName("Frisbee time");
+        trUpdateExercise.setExerciseDescription("Playing at the beach");
+        trUpdateExercise.setOriginalStartDateTime(DateTime.Builder.buildFullString("2020-05-04T10:00:00"));
+        trUpdateExercise.setStartDateTime(DateTime.Builder.buildFullString("2020-05-04T10:00:00"));
+        trUpdateExercise.setEndDateTime(DateTime.Builder.buildFullString("2020-05-04T11:00:00"));
+        trUpdateExercise.execute();
+
+        assertEquals("Should update the exercise", "[{Frisbee time, Playing at the beach, 2020-05-04T10:00:00, "
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Should update the exercise" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5561369)